fl4health/model_bases/autoencoders_base.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABC, abstractmethod
+from typing import Optional, Tuple
+
+import torch
+import torch.nn as nn
+
+from fl4health.utils.dataset_converter import AutoEncoderDatasetConverter
+
+
+class AbstractAe(nn.Module, ABC):
+    """The base class for all Encoder-Decoder based models.
```

Comment:
Super minor: This is a base class for all auto-encoder models rather than all "Encoder-Decoder" based models, as that would include things like encoder-decoder LMs 🙂